network/src/noise_wrapper/handshake.rs

```diff
@@ -87,7 +87,7 @@ impl NoiseWrapper {
         &self,
         socket: TSocket,
         origin: ConnectionOrigin,
-        anti_replay_timestamps: Option<Arc<RwLock<AntiReplayTimestamps>>>,
+        anti_replay_timestamps: Option<&Arc<RwLock<AntiReplayTimestamps>>>,
```

Comment:
It seems like we should introduce this lock into the NoiseWrapper itself, no?

Reply:
once this is pushed I can probably simplify the handshake interface : o

Reply:
How would you simplify it?

Reply:
there's a lot of stuff that's unnecessary just because I didn't want to copy/paste it accross all the transport functions. For one we don't need an `upgrade_connection()` function and we should be able to directly call `dial()` and `accept()`

Reply:
this is what i was thinking about, but left out b/c PR is already huge

```rust
struct NoiseUpgrader {
    config: NoiseConfig,
    anti_replay_timestamps: Option<Arc<RwLock<AntiReplayTimestamps>>>,
    trusted_peers: Option<Arc<RwLock<HashMap<PeerId, NetworkPublicKeys>>>>,
}
```